x-pack/filebeat/input/azureeventhub/eph.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package azureeventhub
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-event-hubs-go/v2"
+	"github.com/Azure/azure-event-hubs-go/v2/eph"
+	"github.com/Azure/azure-event-hubs-go/v2/storage"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// runWithEPH will consume ingested events using the Event Processor Host (EPH) https://github.com/Azure/azure-event-hubs-go#event-processor-host, https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-event-processor-host
+func (a azureInput) runWithEPH() error {
+	// create a new Azure Storage Leaser / Checkpointer
+	cred, err := azblob.NewSharedKeyCredential(a.config.SAName, a.config.SAKey)
+	if err != nil {
+		return err
+	}
+	leaserCheckpointer, err := storage.NewStorageLeaserCheckpointer(cred, a.config.SAName, a.config.SAContainer, azure.PublicCloud)
+	if err != nil {
+		return err
+	}
+	// adding a nil EventProcessorHostOption will break the code, this is why a condition is added and a.processor is assigned
+	if a.config.ConsumerGroup != "" {
+		a.processor, err = eph.NewFromConnectionString(
+			a.workerCtx,
+			fmt.Sprintf("%s%s%s", a.config.ConnectionString, eventHubConnector, a.config.EventHubName),
+			leaserCheckpointer,
+			leaserCheckpointer,
+			eph.WithConsumerGroup(a.config.ConsumerGroup))
+	} else {
+		a.processor, err = eph.NewFromConnectionString(
+			a.workerCtx,
+			fmt.Sprintf("%s%s%s", a.config.ConnectionString, eventHubConnector, a.config.EventHubName),
+			leaserCheckpointer,
+			leaserCheckpointer)
```

Comment:
this code could be common,  passing the `eph.WithConsumerGroup(a.config.ConsumerGroup)` part as a variable